TestAccumulateSummary is unstable

Failed on CI: https://app.circleci.com/pipelines/github/open-telemetry/opentelemetry-collector/6474/workflows/3d38eaf7-5582-4f40-bc9c-1843ac47e0da/jobs/74653

```
=== RUN   TestAccumulateSummary/Summary_with_single_point/WithTimestamp
    collector_test.go:598: 
        	Error Trace:	collector_test.go:598
        	Error:      	Not equal: 
        	            	expected: []*io_prometheus_client.Quantile{(*io_prometheus_client.Quantile)(0xc0002ff020), (*io_prometheus_client.Quantile)(0xc0002ff050)}
        	            	actual  : []*io_prometheus_client.Quantile{(*io_prometheus_client.Quantile)(0xc0002ff1a0), (*io_prometheus_client.Quantile)(0xc0002ff1d0)}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,2 +1,10 @@
        	            	 ([]*io_prometheus_client.Quantile) (len=2) {
        	            	+ (*io_prometheus_client.Quantile)({
        	            	+  Quantile: (*float64)(0.5),
        	            	+  Value: (*float64)(190),
        	            	+  XXX_NoUnkeyedLiteral: (struct {}) {
        	            	+  },
        	            	+  XXX_unrecognized: ([]uint8) <nil>,
        	            	+  XXX_sizecache: (int32) 0
        	            	+ }),
        	            	  (*io_prometheus_client.Quantile)({
        	            	@@ -8,10 +16,2 @@
        	            	   XXX_sizecache: (int32) 0
        	            	- }),
        	            	- (*io_prometheus_client.Quantile)({
        	            	-  Quantile: (*float64)(0.5),
        	            	-  Value: (*float64)(190),
        	            	-  XXX_NoUnkeyedLiteral: (struct {}) {
        	            	-  },
        	            	-  XXX_unrecognized: ([]uint8) <nil>,
        	            	-  XXX_sizecache: (int32) 0
        	            	  })
        	Test:       	TestAccumulateSummary/Summary_with_single_point/WithTimestamp
=== RUN   TestAccumulateSummary/Summary_with_single_point
    collector_test.go:598: 
        	Error Trace:	collector_test.go:598
        	Error:      	Not equal: 
        	            	expected: []*io_prometheus_client.Quantile{(*io_prometheus_client.Quantile)(0xc0002ff020), (*io_prometheus_client.Quantile)(0xc0002ff050)}
        	            	actual  : []*io_prometheus_client.Quantile{(*io_prometheus_client.Quantile)(0xc00073fc20), (*io_prometheus_client.Quantile)(0xc00073fc50)}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,2 +1,10 @@
        	            	 ([]*io_prometheus_client.Quantile) (len=2) {
        	            	+ (*io_prometheus_client.Quantile)({
        	            	+  Quantile: (*float64)(0.5),
        	            	+  Value: (*float64)(190),
        	            	+  XXX_NoUnkeyedLiteral: (struct {}) {
        	            	+  },
        	            	+  XXX_unrecognized: ([]uint8) <nil>,
        	            	+  XXX_sizecache: (int32) 0
        	            	+ }),
        	            	  (*io_prometheus_client.Quantile)({
        	            	@@ -8,10 +16,2 @@
        	            	   XXX_sizecache: (int32) 0
        	            	- }),
        	            	- (*io_prometheus_client.Quantile)({
        	            	-  Quantile: (*float64)(0.5),
        	            	-  Value: (*float64)(190),
        	            	-  XXX_NoUnkeyedLiteral: (struct {}) {
        	            	-  },
        	            	-  XXX_unrecognized: ([]uint8) <nil>,
        	            	-  XXX_sizecache: (int32) 0
        	            	  })
        	Test:       	TestAccumulateSummary/Summary_with_single_point
--- FAIL: TestAccumulateSummary (0.00s)
    --- FAIL: TestAccumulateSummary/Summary_with_single_point/WithTimestamp (0.00s)
    --- FAIL: TestAccumulateSummary/Summary_with_single_point (0.00s)
```

--- a/exporter/prometheusexporter/collector_test.go
+++ b/exporter/prometheusexporter/collector_test.go
@@ -498,30 +498,21 @@ func TestAccumulateSummary(t *testing.T) {
 		vqpN.SetValue(value)
 		return vqpN
 	}
-	quantilesFromMap := func(qf map[float64]float64) (qL []*io_prometheus_client.Quantile) {
-		f64Ptr := func(v float64) *float64 { return &v }
-		for quantile, value := range qf {
-			qL = append(qL, &io_prometheus_client.Quantile{
-				Quantile: f64Ptr(quantile), Value: f64Ptr(value),
-			})
-		}
-		return qL
-	}
 	tests := []struct {
 		name          string
 		metric        func(time.Time) pdata.Metric
 		wantSum       float64
 		wantCount     uint64
-		wantQuantiles []*io_prometheus_client.Quantile
+		wantQuantiles map[float64]float64
 	}{
 		{
 			name:      "Summary with single point",
 			wantSum:   0.012,
 			wantCount: 10,
-			wantQuantiles: quantilesFromMap(map[float64]float64{
+			wantQuantiles: map[float64]float64{
 				0.50: 190,
 				0.99: 817,
-			}),
+			},
 			metric: func(ts time.Time) (metric pdata.Metric) {
 				sp := pdata.NewSummaryDataPoint()
 				sp.SetCount(10)
@@ -595,7 +586,12 @@ func TestAccumulateSummary(t *testing.T) {
 					s := *pbMetric.Summary
 					require.Equal(t, tt.wantCount, *s.SampleCount)
 					require.Equal(t, tt.wantSum, *s.SampleSum)
-					require.Equal(t, tt.wantQuantiles, s.Quantile)
+					// To ensure that we can compare quantiles, we need to just extract their values.
+					gotQuantiles := make(map[float64]float64)
+					for _, q := range s.Quantile {
+						gotQuantiles[q.GetQuantile()] = q.GetValue()
+					}
+					require.Equal(t, tt.wantQuantiles, gotQuantiles)
 				}
 				require.Equal(t, 1, n)
 			})

